From 90a4d6df0259ea249eae066b39c81d969c4fcc77 Mon Sep 17 00:00:00 2001
From: Rene Kita <mail@rkta.de>
Date: Sat, 9 Apr 2022 20:20:25 +0200
Subject: [PATCH 150/172] Remove code duplication

The only difference between allocStr and Strnew_charp_n is that the
former returns NULL if the passed pointer is NULL, while Strnew_charp_n
returns an empty string.
---
 indep.c |   13 +------------
 1 file changed, 1 insertion(+), 12 deletions(-)

--- a/indep.c
+++ b/indep.c
@@ -98,22 +98,11 @@ bzero(void *ptr, int len)
 char *
 allocStr(const char *s, int len)
 {
-    char *ptr;
-
     if (s == NULL)
 	return NULL;
     if (len < 0)
 	len = strlen(s);
-    if (len < 0 || len >= STR_SIZE_MAX)
-	len = STR_SIZE_MAX - 1;
-    ptr = NewAtom_N(char, len + 1);
-    if (ptr == NULL) {
-	fprintf(stderr, "fm: Can't allocate string. Give me more memory!\n");
-	exit(-1);
-    }
-    bcopy(s, ptr, len);
-    ptr[len] = '\0';
-    return ptr;
+    return Strnew_charp_n(s, len)->ptr;
 }
 
 int
